include/WCSimRootEvent.hh

```diff
@@ -40,6 +41,7 @@ private:
   Float_t fStop[3];
   Float_t fStart[3];
   Int_t fParenttype;
+  std::string fCreatorProcess;
```

Comment:
I am a bit concerned about file size using string.  Is it possible to make it an `enum`

Reply:
Expanding on this, it is!

The creator process name is saved in `WCSimTrajectory` by calling `G4VProcess::GetProcessName()` which returns a `G4String`
There's a similar `G4VProcess::GetProcessType()` which returns a `G4ProcessType`, which is an `enum`

```C++
enum G4ProcessType
{
  fNotDefined,
  fTransportation,
  fElectromagnetic,
  fOptical,
  fHadronic,
  fPhotolepton_hadron,
  fDecay,
  fGeneral,
  fParameterisation,
  fUserDefined,
  fParallel,
  fPhonon,
  fUCN
};
```

Reply:
I tried this,
`  if (aTrack->GetCreatorProcess() != 0 )
    {
      const G4VProcess* tempproc = aTrack->GetCreatorProcess();
      creatorProcess = tempproc->GetProcessType();
    }
`

But know I cannot see the names of the processes in my DataFrame, I just see an empty list. I need an extra step here or is this an issue of how I extract the information from the output.root file?

Reply:
`enum` is basically an integer, saving integer as a string will probably give you the wrong result

Reply:
Done